src/mediapipe_internal/mediapipegraphconfig.cpp

```diff
@@ -30,4 +46,75 @@ void MediapipeGraphConfig::setSubconfigPath(const std::string& subconfigPath) {
     FileSystem::setPath(this->subconfigPath, subconfigPath, this->basePath);
 }
 
+bool MediapipeGraphConfig::isReloadRequired(const MediapipeGraphConfig& rhs) const {
+    // Checking OVMS configuration part
+    if (this->graphName != rhs.graphName) {
+        SPDLOG_DEBUG("MediapipeGraphConfig {} reload required due to name mismatch", this->graphName);
+        return true;
+    }
+    if (this->basePath != rhs.basePath) {
+        SPDLOG_DEBUG("MediapipeGraphConfig {} reload required due to basePath mismatch", this->graphName);
+        return true;
+    }
+    if (this->graphPath != rhs.graphPath) {
+        SPDLOG_DEBUG("MediapipeGraphConfig {} reload required due to graphPath mismatch", this->graphName);
+        return true;
+    }
+    if (this->subconfigPath != rhs.subconfigPath) {
+        SPDLOG_DEBUG("MediapipeGraphConfig {} reload required due to subconfigPath mismatch", this->graphName);
+        return true;
+    }
+    // Checking if graph pbtxt has been modified
+    if (currentGraphPbTxtMD5 != "") {
+        std::string newGraphPbTxtMD5 = FileSystem::getFileMD5(rhs.graphPath);
+        if (newGraphPbTxtMD5 != currentGraphPbTxtMD5) {
+            SPDLOG_DEBUG("MediapipeGraphConfig {} reload required due to graph definition modification", this->graphName);
+            return true;
+        }
+    }
+    return false;
+}
+
+Status MediapipeGraphConfig::parseNode(const rapidjson::Value& v) {
+    try {
+        this->setGraphName(v["name"].GetString());
+        if (v.HasMember("base_path")) {
+            std::string providedBasePath(v["base_path"].GetString());
+            if (providedBasePath.back() == '/')
+                this->setBasePath(providedBasePath);
+            else
+                this->setBasePath(providedBasePath + "/");
+        } else {
+            if (!getRootDirectoryPath().empty()) {
+                this->setBasePathWithRootPath();
+                SPDLOG_DEBUG("base_path not defined in config so it will be set to default based on main config directory: {}", this->getBasePath());
+            } else {
+                SPDLOG_ERROR("Mediapipe {} root directory path is not set.", getGraphName());
+                return StatusCode::INTERNAL_ERROR;
+            }
+        }
+        if (v.HasMember("graph_path")) {
+            this->setGraphPath(v["graph_path"].GetString());
+        } else {
+            this->setGraphPath(DEFAULT_GRAPH_FILENAME);
+            SPDLOG_DEBUG("graph_path not defined in config so it will be set to default based on base_path and graph name: {}", this->getGraphPath());
+        }
+        this->setCurrentGraphPbTxtMD5(FileSystem::getFileMD5(this->graphPath));
```

Comment:
Is it possible we have TOCTOU issue here? What I mean is that we do not lock resources when we check MD5, there is possible timeframe where:
1) We load graph configuration - calculate file MD5, save the MD5
2) The file changes
3) We continue loading the file, we validate the graph in `validateForConfigLoadableness` - it reads file from the disk (here it can have different content than in 1)
4) We have saved incorrect MD5, for different file other than the file used by executor

Reply:
?